src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -320,7 +463,233 @@ private void updateWorkOrders()
+        @NotNull final List<Block> freeBlocks = townHall.getColony().getFreeBlocks();
+        @NotNull final List<BlockPos> freePositions = townHall.getColony().getFreePositions();
+
+        freeBlocksList = findPaneOfTypeByID(LIST_FREE_BLOCKS, ScrollingList.class);
+        freeBlocksList.setDataProvider(new ScrollingList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return freeBlocks.size() + freePositions.size();
+            }
+
+            @Override
+            public void updateElement(final int index, @NotNull final Pane rowPane)
+            {
+                if (index < freeBlocks.size())
+                {
+                    rowPane.findPaneOfTypeByID("name", Label.class).setLabelText(freeBlocks.get(index).getRegistryName().toString());
+                }
+                else
+                {
+                    final BlockPos pos = freePositions.get(index - freeBlocks.size());
+                    rowPane.findPaneOfTypeByID("name", Label.class).setLabelText(pos.getX() + " " + pos.getY() + " " + pos.getZ());
+                }
+            }
+        });
+    }
+
+    /**
+     * Called when the "addBlock" button has been triggered.
+     * Tries to add the content of the input field as block or position to the colony.
+     */
+    private void addBlock()
+    {
+        final TextField input = findPaneOfTypeByID(INPUT_BLOCK_NAME, TextField.class);
+        final String inputText = input.getText();
+
+        final Block block = Block.getBlockFromName(inputText);
+
+        if (block != null)
+        {
+            townHall.getColony().addFreeBlock(block);
+            MineColonies.getNetwork().sendToServer(new ChangeFreeToInteractBlockMessage(townHall.getColony(), block, ChangeFreeToInteractBlockMessage.MessageType.ADD_BLOCK));
+        }
+
+        final BlockPos pos = BlockPosUtil.getBlockPosOfString(inputText);
+
+        if (pos != null)
+        {
+            MineColonies.getNetwork().sendToServer(new ChangeFreeToInteractBlockMessage(townHall.getColony(), pos, ChangeFreeToInteractBlockMessage.MessageType.ADD_BLOCK));
+            townHall.getColony().addFreePosition(pos);
+        }
+
+        fillFreeBlockList();
+        input.setText("");
+    }
+
+    /**
+     * Called when the permission button has been triggered.
+     *
+     * @param button the triggered button.
+     */
+    private void trigger(@NotNull final Button button)
+    {
+        @NotNull final Pane pane = button.getParent().getChildren().get(2);
+        int index = 0;
+        if (pane instanceof Label)
+        {
+            index = Integer.valueOf(((Label) pane).getLabelText());
+        }
+        final boolean trigger = LanguageHandler.format(ON).equals(button.getLabel());
+        final Permissions.Action action = Permissions.Action.values()[index];
+        final Permissions.Rank rank = Permissions.Rank.valueOf(actionsList.getParent().getID().toUpperCase());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Define the locale to be used in this String operation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1449)